src/gpio.rs

```diff
@@ -518,6 +538,51 @@ macro_rules! gpio {
                         Ok(unsafe { (*$GPIOX::ptr()).idr.read().bits() & (1 << $i) == 0 })
                     }
                 }
+
+                impl<MODE> ExtiPin for $PXi<Input<MODE>> {
+                    /// Configure EXTI Line $i to trigger from this pin.
+                    fn make_interrupt_source(&mut self, afio: &mut AFIO) {
+                        let offset = 4 * ($i % 4);
+                        afio.$exticri.modify(|r, w| unsafe {
+                            let mut exticr = r.bits();
+                            exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                            w.bits(exticr)
```

Comment:
There already exist access methods for ```AFIO_EXTICR``` registers. For instance, example for EXTI10_15 w/o macros woud look like this:
```
        // select PD15 as source input for line EXTI10_15
        afio.exticr4
            .exticr4()
            .modify(|_, w| unsafe { w.exti15().bits(0b0011) });
```
So is it possible to simplify suggested implementation a bit using ```exti``` accessors ?

Reply:
The same as for the `imr, rtsr, ftsr` registers applies here: It would be possible to use these accessors, but at the cost of having to repeat the pin number a couple of times in each of the n*16 pin definitions (n being the number of ports), as `imrN`, `rtsrN`, `exticrM` and so on.
I don't think it would simplify things.